### ISSUE TITLE
How can I get one resource to update another based on id in Rails?

### ISSUE THREAD

**Issue by lisawilliams** [ga-wdi-boston/capstone-project#771]
Esteemed consultants and 🐤s, 

I now have three active resources in my Rails API: `users`, `shows`, and `artworks`. 

Artworks references `shows` -- here's the schema for it: 

```
  create_table "artworks", force: :cascade do |t|
    t.string   "title"
    t.string   "artist"
    t.integer  "yearcreated"
    t.string   "artform"
    t.string   "materials"
    t.string   "surface"
    t.string   "dimensions"
    t.integer  "show_id"
    t.datetime "created_at",  null: false
    t.datetime "updated_at",  null: false
    t.index ["show_id"], name: "index_artworks_on_show_id", using: :btree
  end
```

I am now updating the controller generated by the `generate scaffold` command.  

`artworks controller` create function currently reads like this:  

```
  # POST /artworks
  def create
    @artwork = current_user.shows.artwork.build(artwork_params)

    if @artwork.save
      render json: @artwork, status: :created, location: @artwork
    else
      render json: @artwork.errors, status: :unprocessable_entity
    end
  end
```

I know that it needs to update a *specific* show based on the show_id.  But how do I write that?

**Comment by lisawilliams** [comment]
^^^ referenced commit for ease of browsing ^^^

**Comment by scottyscripts** [comment]
You are definitely on the right track with creating artwork. We must do `@artwork = current_user.shows.artwork.build` because artwork belongs to a show which belongs to a user. However, instead of using `current_user.shows`, we want to select a specific show. 
For example, we want to find an instance of the show we want to create the artwork for.
so something like
``` 
@show = Show.find(11)
current_user.show.artwork.build(   etc...
```

**Comment by lisawilliams** [comment]
Yes, please, let's do it!  That is, I would appreciate some help debugging this.

**Comment by lisawilliams** [comment]
This looks like it might be useful -- I am reading it now: 

http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html

**Comment by lisawilliams** [comment]
Okay, @sdavidson140 took a look at this issue via adding a binding.pry statement to let us debug it in the Rails console, and we broke it down into smaller steps: 

1) capture show_id from params
  show_id = params[:artwork][:show_id]
  
2) select a show that belongs to current user
  current_show = current_user.shows.find(show_id)
  
3) create artwork from selected show
  current_show.artworks.build(artwork_params)

I'm going to take what we discovered in Rails Console and add it to artworks_controller.rb and test it with a CURL script and then close this issue.  I will append commits to this issue so future students and consultants can see the solution code.

**Comment by lisawilliams** [comment]
The commit message from the commit above ^^^^^ 

   Wrote `create` function in artworks_controller
    
    Added code to allow us to associate an artwork with a show.
    Added curl script to create an artwork and tested curl script.
    
    Script returns `201 Created` as expected.
    Confirmed that artwork created is in the `artworks` table in the
    database with the correct values and correct associated show.
    
    Closes Issue #36.
    
    Closes Capstone Issue 771.
    https://github.com/ga-wdi-boston/capstone-project/issues/771

Thanks @sdavidson140 !
